src/main/java/org/mvel2/util/ParseTools.java

```diff
@@ -79,6 +81,13 @@
   public static final Object[] EMPTY_OBJ_ARR = new Object[0];
   public static final Class[] EMPTY_CLS_ARR = new Class[0];
 
+  private static final Map<String, Class<?>> CLASS_MAP = new ConcurrentHashMap<String, Class<?>>();
```

Comment:
Using global object is not a good pattern. It introduces unpredictability due to statefulness.